### DISCUSSION TITLE
Why adding space to slint file, cause running 8s build script?

### DISCUSSION THREAD

**Discussion by qarmin** [slint-ui/slint#4215] [Q&A]
After first full `cargo check` build, adding single space in slint file and then running again `cargo check` takes 30 seconds.

~20 seconds is related with rust, but ~7 seconds comes from build script

A little strange that most of the time takes `std::io::Write::write_fmt` inside `compile_with_config` function and when looking at flamegraph I see hundreds of operations on files.

Is this normal behavior?

![Screenshot from 2023-12-25 15-11-55](https://github.com/slint-ui/slint/assets/41945903/1a66c4d8-8cf7-4877-bb40-2c7155153f21)

**Comment by qarmin** [comment]
So I think that here should be used buffered file operation, which should decrease time 

https://github.com/slint-ui/slint/blob/d1529af3cc079c724fcbe1a4101a14013eda8b76/api/rs/build/lib.rs#L357-L358

**Comment by qarmin** [comment]
Simple changing
```
 let file = std::fs::File::create(&output_file_path).map_err(CompileError::SaveError)?; 
 let mut code_formatter = CodeFormatter::new(file); 
```
into
```
    let file = std::fs::File::create(&output_file_path).map_err(CompileError::SaveError)?;
    let buffer = std::io::BufWriter::new(file);
    let mut code_formatter = CodeFormatter::new(buffer);
```
decreased build script time from ~8s to 2s

**Reply by ogoffart** [reply]
Thanks a lot for doing the profiling and figuring out the problem.

Someone did contribute a change in https://github.com/slint-ui/slint/pull/4217
